src/gz21_ocean_momentum/trainScript.py

```diff
@@ -161,13 +165,39 @@ def check_str_is_None(string_in: str):
 )
 params = parser.parse_args()
 
+def argparse_get_mlflow_artifact_path_or_direct_or_fail(
+        mlflow_artifact_name: str, params: dict[str, Any]
+        ) -> str:
+    """Obtain a filepath either from an MLflow run ID and artifact name, or a
+    direct path if provided.
+
+    params must have keys run_id and forcing_data_path.
+
+    Only one of run_id and path should be non-None.
+
+    Note that the filepath is not checked for validity (but for run_id, MLflow
+    probably will assert that it exists).
+
+    Effectful: errors result in immediate program exit.
+    """
+    if params.run_id is not None and params.run_id != "None":
+        if params.forcing_data_path is not None and params.forcing_data_path != "None":
+            # got run ID and direct path: bad
+            raise TypeError("overlapping options provided (--forcing-data-path and --exp-id)")
+
+        # got only run ID: obtain path via MLflow
+        mlflow.log_param("source.run-id", params.run_id)
+        mlflow_client = mlflow.tracking.MlflowClient()
+        return mlflow_client.download_artifacts(params.run_id, mlflow_artifact_name)
+
+    if params.forcing_data_path is not None and params.forcing_data_path != "None":
+        # got only direct path: use
+        return params.forcing_data_path
```

Comment:
Just so that I am sure to understand: In this case we do not need to bother with mlflow, right?

Reply:
In this case (where `--forcing-data-path` is provided), MLflow isn't used to locate the forcing data. You can run the script with or without MLflow regardless of CLI options used.